From 003d0a6259eb8a0c8809c051ff10880d9c4bbb56 Mon Sep 17 00:00:00 2001
From: Ivan Shvedunov <ivan4th@gmail.com>
Date: Sat, 16 Jan 2021 03:43:01 +0300
Subject: [PATCH] tcp: remove bad assert

Don't assume that half-open connections pending removal are always
successful.

Type: fix
Signed-off-by: Ivan Shvedunov <ivan4th@gmail.com>
Change-Id: I955077a4ed2389b9ee38d97e27a7c7761c860a4a
---
 src/vnet/tcp/tcp_input.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/src/vnet/tcp/tcp_input.c b/src/vnet/tcp/tcp_input.c
index a86161c6d..aab52bbb0 100644
--- a/src/vnet/tcp/tcp_input.c
+++ b/src/vnet/tcp/tcp_input.c
@@ -1814,13 +1814,10 @@ tcp46_syn_sent_inline (vlib_main_t * vm, vlib_node_runtime_t * node,
 	  goto drop;
 	}
 
-      /* Half-open completed recently but the connection was't removed
-       * yet by the owning thread */
+      /* Half-open completed or cancelled recently but the connection
+       * was't removed yet by the owning thread */
       if (PREDICT_FALSE (tc0->flags & TCP_CONN_HALF_OPEN_DONE))
 	{
-	  /* Make sure the connection actually exists */
-	  ASSERT (tcp_lookup_connection (tc0->c_fib_index, b0,
-					 my_thread_index, is_ip4));
 	  error0 = TCP_ERROR_SPURIOUS_SYN_ACK;
 	  goto drop;
 	}
-- 
2.28.0

